framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -6641,6 +6641,78 @@ VkResult VulkanReplayConsumerBase::OverrideGetPhysicalDeviceToolProperties(
     }
 }
 
+VkResult
+VulkanReplayConsumerBase::OverrideWaitSemaphores(PFN_vkWaitSemaphores func,
+                                                 VkResult             original_result,
+                                                 const DeviceInfo*    device_info,
+                                                 const StructPointerDecoder<Decoded_VkSemaphoreWaitInfo>* pInfo,
+                                                 uint64_t                                                 timeout)
+{
+    assert((device_info != nullptr) && (pInfo != nullptr) && !pInfo->IsNull() && (pInfo->GetPointer() != nullptr));
+    VkDevice                   device    = device_info->handle;
+    const VkSemaphoreWaitInfo* wait_info = pInfo->GetPointer();
+    VkResult                   result;
+
+    // If expected result is VK_SUCCESS, ensure that vkWaitSemaphores waits until semaphores
+    // are available by using a timeout of UINT64_MAX.
+    // If expected result is VK_TIMEOUT, try to get a timeout by using a timeout of 0.
+    // If expected result is anything else, use the passed in timeout value.
+    if (original_result == VK_SUCCESS)
+        timeout = std::numeric_limits<uint64_t>::max();
+    else if (original_result == VK_TIMEOUT)
+        timeout = 0;
```

Comment:
Does this guarantee timeout is returned?  Or should we also return TIMEOUT?

Reply:
No, this doesn't guarantee a VK_TIMEOUT is returned. This is the case in which the capture returned VK_TIMEOUT, but for whatever reason, the replay returns a different value (maybe the timing in the replay is different so that a timeout doesn't occur, or the device is now in a bad state and the replay returns VK_ERROR_DEVICE_LOST). The calling code will print out a warning that the return value didn't match the capture. I'm not sure that we can replay correctly given that a timeout occurred during capture. This behavior is the same as in the replay of vkWaitForFence. I propose we leave the code as submitted in this PR.